src/encoded/static/components/form.js

```diff
@@ -890,6 +891,13 @@ export class Form extends React.Component {
         return this.state.isDirty && this.state.isValid && !this.state.editor_error && !this.communicating;
     }
 
+    cancel() {
+        const link = url.parse(this.context.location_href, true);
+
+        // the last '/' is a hack to reload the page. In app.js, fallbackNavigate() did not without it.
+        this.context.navigate(`${link.pathname}/`, { reload: true });
```

Comment:
Thanks for that comment. It looks wrong to add the slash without it.

Reply:
In fallbackNavigate() (in app.js), window.location.assign would not redirect properly without it. I have no idea why. That is why I did this. I also did not want to add too much internal details about app.js in form.js and other files.